src/main/java/org/jboss/tm/listener/event/TransactionLifecycleEventType.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.tm.listener.event;
+
+import java.util.EnumSet;
+
+/**
+ * A mechanism for providing extensible enum types.
+ *
+ * @param <E> the Enum type of the implementing class - ie the implementation will have all of the methods of
+ *           Enum<E>
+ */
+public interface TransactionLifecycleEventType<E extends Enum<E>> {
```

Comment:
```java
public interface TransactionLifecycleEventType<E extends Enum<E>>
```
is an interface and all concrete event types must implement it - ie AssociationLifecycleEventType implements. TransactionLifecycleEventType

The reason for this approach is twofold:
- it ensures that all event types are enums
- it is a generic type so that I can ensure that the rest of the framework is typesafe whilst still allowing for the introdution of new types